gcp/workers/linter/linter.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Run osv-linter on all OSV records."""
+
+import json
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+from google.cloud import ndb, storage
+
+import osv.logs
+from osv.models import utcnow
+
+# The Go binary is copied to /usr/local/bin in the Dockerfile
+OSV_LINTER_PATH = '/usr/local/bin/osv-linter'
+VULN_BUCKET = 'osv-test-vulnerabilities'
+ZIP_FILE_PATH = 'all.zip'
+TEST_DATA = './test_data'
+GCP_PROJECT = 'oss-vdb-test'
+
+ERROR_CODE_MAPPING = {
+    'A0001': osv.ImportFindings.INVALID_RECORD,
+    'R0001': osv.ImportFindings.INVALID_RANGE,
+    'R0002': osv.ImportFindings.INVALID_RANGE,
+    'P0001': osv.ImportFindings.INVALID_PACKAGE,
+    'P0002': osv.ImportFindings.INVALID_VERSION,
+    'P0003': osv.ImportFindings.INVALID_PURL,
+}
+
+# TODO(gongh@): query the mapping from the SourceRepository database
+# instead of hardcoding it.
+PREFIX_TO_SOURCE = {
+    'ALBA-': 'almalinux-alba',
+    'ALEA-': 'almalinux-alea',
+    'ALSA-': 'almalinux-alsa',
+    'A-': 'android',
+    'ASB-A': 'android',
+    'PUB-A': 'android',
+    'BIT-': 'bitnami',
+    'CGA-': 'chainguard',
+    'CURL-': 'curl',
+    'CVE-': 'cve-osv',
+    'DLA-': 'debian-dla',
+    'DSA-': 'debian-dsa',
+    'DTSA-': 'debian-dtsa',
+    'GHSA-': 'ghsa',
+    'GO-': 'go',
+    'HSEC-': 'haskell',
+    'MGASA-': 'mageia',
+    'MAL-': 'malicious-packages',
+    'MINI-': 'minimos',
+    'OSV-': 'test-oss-fuzz',
+    'PSF-': 'psf',
+    'PYSEC-': 'python',
+    'RSEC-': 'r',
+    'RHBA-': 'redhat',
+    'RHEA-': 'redhat',
+    'RHSA-': 'redhat',
+    'RLSA-': 'rockylinux',
+    'RXSA-': 'rockylinux-rxsa',
+    'RUSTSEC-': 'rust',
+    'openSUSE-': 'suse',
+    'SUSE-': 'suse',
+    'UBUNTU-': 'ubuntu-cve',
+    'LSN-': 'ubuntu-lsn',
+    'USN-': 'ubuntu-usn',
+    'GSD-': 'uvi',
+    'V8-': 'V8',
+}
+
+
+def download_zip(all_zip: str):
+  """Downloads all.zip file from bucket."""
+  storage_client = storage.Client()
+  bucket = storage_client.get_bucket(VULN_BUCKET)
+  try:
+    blob = bucket.blob(ZIP_FILE_PATH)
+    blob.download_to_filename(all_zip)
+  except Exception as e:
+    logging.exception('Failed to download all.zip: %s', e)
```

Comment:
nit: remove all.zip reference, since this function is more general and not specific to downloading all.zip

Reply:
updated function for more general file downloading purpose